src/partialBind.ts

```diff
@@ -0,0 +1,63 @@
+/* eslint-disable @typescript-eslint/no-explicit-any, @typescript-eslint/prefer-readonly-parameter-types, @typescript-eslint/no-unsafe-return --
+ * Function inference doesn't work when `unknown` is used as the parameters
+ * generic type, it **has** to be `any`.
+ */
+import type { IterableContainer, RemedaTypeError } from "./internal/types";
+
+type PartialBindError<Message extends string> = [
+  RemedaTypeError<"partialBind", Message>,
+];
+
+type RemovePrefix<
+  T extends IterableContainer,
+  Prefix extends IterableContainer,
+> = Prefix["length"] extends 0
+  ? T
+  : T["length"] extends 0
+    ? PartialBindError<"Too many args provided to function">
+    : T extends [infer THead, ...infer TRest]
+      ? Prefix extends [infer PrefixHead, ...infer PrefixRest]
+        ? PrefixHead extends THead
+          ? RemovePrefix<TRest, PrefixRest>
+          : PartialBindError<"Argument of the wrong type provided to function">
+        : Prefix extends Array<THead>
+          ? [THead?, ...RemovePrefix<TRest, Prefix>]
+          : never
+      : T extends [(infer THead)?, ...infer TRest]
+        ? Prefix extends [infer PrefixHead, ...infer PrefixRest]
+          ? PrefixHead extends THead
+            ? RemovePrefix<TRest, PrefixRest>
+            : PartialBindError<"Argument of the wrong type provided to function">
+          : Prefix extends TRest
+            ? TRest
+            : never
+        : never;
+
+type PartiallyBound<
+  F extends (...args: any) => any,
+  T extends IterableContainer,
```

Comment:
The usage of T and F here is confusing. `F` makes sense for the main type here, which is the function, but the partial arguments list should have an explicit name.